sootup.core/src/main/java/sootup/core/jimple/common/expr/JNegExpr.java

```diff
@@ -69,17 +69,18 @@ public void toString(@Nonnull StmtPrinter up) {
   public Type getType() {
     Value op = getOp();
 
-    if (op.getType().equals(PrimitiveType.getInt())
-        || op.getType().equals(PrimitiveType.getByte())
-        || op.getType().equals(PrimitiveType.getShort())
-        || op.getType().equals(PrimitiveType.getBoolean())
-        || op.getType().equals(PrimitiveType.getChar())) {
+    // TODO ms: create a TypeVisitor for this -> O(1)
+    if (op.getType() == PrimitiveType.getInt()
```

Comment:
Does == work here? 
because it is the same object?

Reply:
exactly - for thatreason we implemented it this way - and to save memory and as we have private ctors its no other way to get to true - except maybe via nasty reflection stuff making the ctor accessible etc